src/types/utils.cpp

```diff
@@ -378,71 +101,239 @@ til::color Utils::ColorFromHexString(const std::string_view str)
 }
 
 // Routine Description:
-// - Shorthand check if a handle value is null or invalid.
+// - Given a color string, attempts to parse the color.
+//   The color are specified by name or RGB specification as per XParseColor.
 // Arguments:
-// - Handle
+// - string - The string containing the color spec string to parse.
 // Return Value:
-// - True if non zero and not set to invalid magic value. False otherwise.
-bool Utils::IsValidHandle(const HANDLE handle) noexcept
+// - An optional color which contains value if a color was successfully parsed
+std::optional<til::color> Utils::ColorFromXTermColor(const std::wstring_view string) noexcept

```

Comment:
It suddenly hit me that `XParseColor` and XOrg color names are actually two different specs. So I made this to unify them and try to give it the best name I can think of.